src/wavelet_matrix.rs

```diff
@@ -0,0 +1,559 @@
+use anyhow::{anyhow, Result};
+use std::ops::Range;
+
+use crate::{broadword, BitVector, RsBitVector};
+
+/// Time and space efficient index data structures for sequence.
```

Comment:
```suggestion
/// Time and space efficient index data structures for a sequence of integers, supporting some queries such as ranking, selection, and intersection. 
```

It would be better to add more helpful descriptions.